### DISCUSSION TITLE
Design Pattern in C#

### DISCUSSION THREAD

**Discussion by Shervan360** [ProfessionalCSharp/ProfessionalCSharp2021#138] [Ideas]
Hello,

Isn't it better to have a specific chapter for Design Pattern in C#? I think your book needs a chapter for design patterns in C# with examples.

**Comment by christiannagel** [comment]
It would be great to have an additional chapter on design patterns and how to implement it with .NET. However, there's a limit on the page numbers. There are also books available just dedicated to *design patterns*. 

Anyway, some design patterns are covered in the book. For example, **dependency injection** is covered in chapter 4 (*Object-Oriented Programming*) and 15 (*Dependency Injection and Configuration*), the **MVVM pattern** is covered in chapter 30 (*Patterns with XAML Apps*)
